lib/sidekiq/form526_job_status_tracker/job_tracker.rb

```diff
@@ -20,51 +20,55 @@ module JobTracker
         # @param msg [Hash] The message payload from Sidekiq
         #
         # rubocop:disable Metrics/MethodLength
-        def job_exhausted(msg, statsd_key_prefix)
+        def job_exhausted(msg, statsd_key)
           job_id = msg['jid']
+          job_class = msg['class'].demodulize
           error_class = msg['error_class']
           error_message = msg['error_message']
           timestamp = Time.now.utc
           form526_submission_id = msg['args'].first
 
-          values = {
-            form526_submission_id:,
-            job_id:,
-            job_class: msg['class'].demodulize,
-            status: Form526JobStatus::STATUS[:exhausted],
-            error_class:,
-            error_message:,
-            bgjob_errors: {},
-            updated_at: timestamp
-          }
-
           form_job_status = Form526JobStatus.find_by(job_id:)
           bgjob_errors = form_job_status.bgjob_errors || {}
           new_error = {
             "#{timestamp.to_i}": {
               caller_method: __method__.to_s,
-              error_class:, error_message:,
-              timestamp:
+              error_class:,
+              error_message:,
+              timestamp:,
+              form526_submission_id:
             }
           }
-          bgjob_errors.merge!(new_error)
-          values[:bgjob_errors] = bgjob_errors
 
-          JobTracker.send_backup_submission_if_enabled(form526_submission_id:, job_class: values[:job_class], job_id:,
+          JobTracker.send_backup_submission_if_enabled(form526_submission_id:, job_class:, job_id:,
                                                        error_message:, error_class:)
 
           # rubocop:disable Rails/SkipsModelValidations
-          Form526JobStatus.upsert(values, unique_by: :job_id)
+          form_job_status.update(
+            status: Form526JobStatus::STATUS[:exhausted],
+            bgjob_errors: bgjob_errors.merge(new_error)
+          )
```

Comment:
double check upsert isn't doing anything we need?

Reply:
Because the `upsert` method is being called on the `Form526JobStatus` record that is unique to the `job_id` already used to find the previously defined `form_job_status` variable, I thought an update of the latter seemed reasonable following the process of how updates are being homogenized in all the other exhaustion hooks.

One instance where `upsert` would be preferred is if a record didn't already exist for the given `job_id` and a new record had to be inserted. But, in this case, I thought a `Form526JobStatus` record, along with a `job_id` would have been created along with the associated `Form526Submission` record creation. I'm curious to get your perspective, however.